### ISSUE TITLE
Question: Cookie jar support?

### ISSUE THREAD

**Issue by Narzerus** [tomas/needle#101]
Does this library have any sort of cookie jar support. Lets say I wanted to save multiple jars in my app and use them when I need them.

Is that possible here?

**Comment by tomas** [comment]
Actually, just yesterday cookie support was added into Needle. This is what you can do.

``` js
var needle = require('needle');

var cookies = { foo: 'bar', user: 'someone' };

// by passing the object in the cookies option, Needle will set them in the 'Cookies' request header.
needle.get('foobar.com/path', { cookies: cookies }, function(err, resp) {
  if (!err) console.log(resp.cookies); // will spit out an object with the response cookies (by parsing the set-cookie header)
})

// additionally, you can also ask needle to persist cookies on redirects by setting the follow_set_cookies option to true. this obviously requires the follow_max count to a number higher than one.
needle.get('foobar.com/path', { follow_max: 3, follow_set_cookies: true }, function(err, resp) {
  // if a redirect was followed, needle will parse and set the cookies accordingly
  // the resp.cookies object will be ones received on the last response
})
```

I hope this helps.

**Comment by Narzerus** [comment]
Hmmm, aight, is the `cookies` POJO updated after the requests are done. Example:

```javascript
  var cookies = {foo: 'bar'};

  need.get('foobar.com/path', {cookies: cookies}, function (err, resp) {
    // Lets imagine the response headers had a Set-Cookie: new=cookie
    // Will this be the output?
    console.log(cookies); // {foo: 'bar', new: 'cookie'}
  });
```

Btw, I was looking for a request replacement, because the way they handle cookies is hideous and just plain wrong when you need to access them manually. This library looks pretty straightforward.


P.S. Are you by any chance from Chile? I think I've atended a talk you did.

**Comment by tomas** [comment]
Yup, from Chile like the porotos.

The cookie object you'll receive as part of the response object are just the response cookies (those contained in the `set-cookie` header). So in that case you should only get `{ new: 'cookie' }` in `resp.cookies`.

**Comment by Narzerus** [comment]
Sure, I meant if the cookies object passed was updated in the process but I believe it doesn't. Btw,
here's a little bug I fixed for you #102

Need any further help with the library aperro cause I think I might stick to it ;)
